### PR TITLE
Update dotnet-install scripts to support installing the ASP.NET Core runtime

### LINKED ISSUE #8252: Update dotnet-install to support installing aspnetcore
We should update dotnet-install.ps1/sh to support installing the Microsoft.AspNetCore.All shared runtime. 

Possible usage syntax:

**Option 1** 
Option with value: `-Runtime <name>`
```
dotnet-install.ps1 -Runtime dotnet
dotnet-install.ps1 -Runtime aspnetcore

dotnet-install.sh --runtime dotnet
dotnet-install.sh --runtime aspnetcore
```
For compatibility, `--shared-runtime` could just become sugar for `--runtime dotnet`

**Option 2** 
New switch: `-AspNetCoreRuntime`
```
dotnet-install.ps1 -AspNetCoreRuntime
dotnet-install.sh --aspnetcore-runtime
```

Invalid usage: mixing -SharedRuntime and -AspNetCoreRuntime. This would be an error.

@johnbeisner @livarcocc - what would be your usage preference? Is there another syntax that might be better?

cc @JunTaoLuo

### PR DESCRIPTION
**Pull request by natemcmaster**
Resolves #8252

Adds support for downloading the aspnetcore shared runtime from the dotnetcli Azure blob feed.

Obsoletes -SharedRuntime in favor of the more-specific `-Runtime <name>` option. 

cc @JunTaoLuo

### CODE REVIEW

File: scripts/obtain/dotnet-install.ps1 [hunk 1]
```diff
@@ -34,8 +34,17 @@
     Architecture of dotnet binaries to be installed.
     Possible values are: <auto>, x64 and x86
 .PARAMETER SharedRuntime
```
[the remark this answers is not in the document]
Comment (natemcmaster) [reply scripts/obtain/dotnet-install.ps1]: Must have. It's there: https://github.com/dotnet/cli/pull/8333/files#diff-3385102241bcd3718b9ca13a0c55b9afR794

**Comment by natemcmaster** [comment]
Thanks @johnbeisner !

**Comment by johnkors** [comment]
Does this include runtime stores, or is there any other way of installing runtime stores?

**Comment by natemcmaster** [comment]
@johnkors this does not include the aspnetcore runtime store. The aspnetcore shared runtime is the replacement for the runtime store in 2.1. We are considering to backporting the shared runtime to 2.0 as well, but in the meantime, you can get the runtime store here: https://github.com/dotnet/core/blob/master/release-notes/download-archives/2.0.5-download.md#server-hosting

**Comment by johnkors** [comment]
Huh, is the runtime store already going away? Or did you mean you moved aspnet core all from the runtime store to some other place?

I was looking into scripting installs of packages we depend on into the store, but if this going away anyways - I might as well wait. What's the plan going fwd?

**Comment by natemcmaster** [comment]
@johnkors I think Damian was working on an announcement for this. We are still working through implementation details on how exactly everything works. So far, these changes only affect 2.1.

Yes https://github.com/aspnet/MetaPackages/issues/248 describe part of what is going one. Here are some other issues for this: https://github.com/aspnet/Universe/issues/721, https://github.com/aspnet/Universe/issues/720, https://github.com/aspnet/Universe/issues/662

**Pull request merged**